src/main/java/net/revelc/code/impsort/maven/plugin/AbstractImpSortMojo.java

```diff
@@ -325,4 +385,49 @@ static LanguageLevel getLanguageLevel(String compliance) {
     return LanguageLevel.valueOf(langLevel);
   }
 
+  /**
+   * Store file hash cache.
+   *
+   * @param props the props
+   */
+  private void storeFileHashCache(Properties props) {
+    File cacheFile = new File(this.cachedir, CACHE_PROPERTIES_FILENAME);
+    try (OutputStream out = new BufferedOutputStream(new FileOutputStream(cacheFile))) {
+      props.store(out, null);
+    } catch (IOException e) {
+      getLog().warn("Cannot store file hash cache properties file", e);
+    }
+  }
+
+  /**
+   * Read file hash cache file.
+   *
+   * @return the properties
+   */
+  private Properties readFileHashCacheFile() {
+    Properties props = new Properties();
+    Log log = getLog();
+    if (!this.cachedir.exists()) {
+      if (!this.cachedir.mkdirs()) {
+        log.warn("Unable to create cache directory '" + this.cachedir.getPath() + "'.");
+      }
+    } else if (!this.cachedir.isDirectory()) {
+      log.warn("Something strange here as the '" + this.cachedir.getPath()
+          + "' supposedly cache directory is not a directory.");
+      return props;
+    }
+
+    File cacheFile = new File(this.cachedir, CACHE_PROPERTIES_FILENAME);
+    if (!cacheFile.exists()) {
+      return props;
+    }
+
+    try (BufferedInputStream stream = new BufferedInputStream(new FileInputStream(cacheFile))) {
+      props.load(stream);
+    } catch (IOException e) {
+      log.warn("Cannot load file hash cache properties file", e);
+    }
+    return props;
+  }
```

Comment:
commons-configuration2 has much simpler API for reading/writing properties files. It's probably worth using that, for maintainability:

```java
    var config = new PropertiesConfiguration();
    // read file
    try (FileReader out = new FileReader(file, UTF_8)) {
      config.read(out);
    }
    // write file
    try (FileWriter out = new FileWriter(file, UTF_8)) {
      config.write(out);
    }
```

Reply:
The first lines are checking the parent directory / cache file existence, so they would be required in both cases.

So if it's only about changing from:
```
try (BufferedInputStream stream = new BufferedInputStream(new FileInputStream(cacheFile))) {
      props.load(stream);
}
```

to 
```
try (FileReader out = new FileReader(file, UTF_8)) {
      config.read(out);
}
```

I'd rather go with the NIO api:
```
    try (BufferedReader reader = Files.newBufferedReader(cacheFile.toPath())) {
      props.load(reader);
    }
```
to avoid the additional dependency.  Also the `FileReader` constructor that takes a `Charset` is defined in JDK >= 11 and the project requirement is still JDK 8 afaik.

Reply:
Okay, seems reasonable.